### PR TITLE
Added ability to set a moderator via JWT and added a Clear tool for moderators

### PR DESCRIPTION
**Pull request by jamesdeacon**
You can now add a `moderator` key within the jwt. If this is set to true then the user will have access to a Clear tool, which will wipe the entire contents of the board.

This fixes issues #92 and #212 

<sub>
By opening a pull request, I certify that I hold the intellectual property of the code I am submitting,
and I am granting the initial authors of WBO a perpetual, worldwide, non-exclusive, royalty-free, and irrevocable license to this code.
</sub>

### CODE REVIEW

**Review by lovasoa — CHANGES_REQUESTED** [review]
Great, thanks for the PR !

A few remarks : 

 - It looks like the server accepts "clear" requests without checking the moderator role.
 - Can we use the standard [role claim](https://www.rfc-editor.org/rfc/rfc7643#section-4.1.2) in th JWT instead of a custom "moderator" claim ?
 - Where does the SVG come from ?

**Comment by jamesdeacon** [comment]
Not sure why the build test is failing. It completes all tests when I run locally and even in the test itself it is passing most of the tests, it's just the circle test that is failing, and I've not touched that code. Are you able to re-run the test?

In response to your requests
- Server checks the token before running the clear command.
- Using standard role claim now
- The SVG is from FlatIcon. I have an unlimited license to use it on websites and in software applications.

File: server/server.js [hunk 1]
```diff
@@ -98,18 +98,27 @@ function validateBoardName(boardName) {
...
     var token = url.searchParams.get("token");
     if(token) {
-      jsonwebtoken.verify(token, config.AUTH_SECRET_KEY);
+      var payload = jsonwebtoken.verify(token, config.AUTH_SECRET_KEY);
+      var roles = payload.roles;
+      if(roles) {
+        return roles.includes("moderator");
+      } else {
+        return false;
+      }
```
Comment (lovasoa) [inline server/server.js:116]: Ca we avoid code duplication and use isModerator here ?

**Review by lovasoa — CHANGES_REQUESTED** [review]
Great ! I'm ready to merge this, but you'll need to change the illustration. You need to have the right of redistributing it under [WBO's license](https://github.com/lovasoa/whitebophir/blob/master/LICENSE). 

You can create your own svg icon, or use an existing true open-source icon, such as one from [remixicon](https://remixicon.com/) for example

**Comment by jamesdeacon** [comment]
I've made the code change requested and have changed the icon to one I've created

**Comment by jamesdeacon** [comment]
Sorry, got confused. I've removed the duplication now. Both server.js and boardData.js now reference the new jwtauth.js file

**Comment by lovasoa** [comment]
Great, thank you for your contribution!

**Comment by jamesdeacon** [comment]
That's great. Thanks so much for the brilliant whiteboard! Happy to contribute

**Comment by mwllgr** [comment]
Will this get published on Docker Hub, @lovasoa? Thanks for the nice feature, @jamesdeacon!

**Pull request merged**